dask_expr/_shuffle.py

```diff
@@ -241,9 +241,7 @@ def _lower(self):
                 ]
             drop_columns = partitioning_index.copy()
         elif index_shuffle:
-            frame = Assign(frame, "_partitions_0", frame.index)
-            partitioning_index = ["_partitions_0"]
-            drop_columns = partitioning_index.copy()
+            dtypes = np.float64 if _is_numeric_cast_type(dtype) else None
```

Comment:
dtype isn't defined here.

Reply:
Thx, I should stop making non-trivial changes in the ui